### DISCUSSION TITLE
Adding Support to linear optical ruler

### DISCUSSION THREAD

**Discussion by muhammadelmogy** [grblHAL/core#396] [Ideas]
I wounder if there would be in the future support for linear optical ruler like the one in the image:
![Screenshot 2023-11-24 153203](https://github.com/grblHAL/core/assets/8148383/1e0e58ce-811f-46d3-b644-cf50d53a5d5a)

**Comment by terjeio** [comment]
For implementing closed loop motion with steppers? Then [this discussion](https://github.com/grblHAL/RP2040/discussions/80) might be of interest.

**Reply by muhammadelmogy** [reply]
I am not entirely sure about how it is being used or how does the core of the code would interact with it (I read some where about it being used with LinuxCNC and MESA hardware)
I am more interested if there would be a plane on implementing this feature in the future because I would make some space in my current new design for it or even buy it and try it later if the feature became available later.
I would love to implement this myself but I do not have enough experience to do it since my main domain is web development !
